### ISSUE TITLE
Question: Too many requests (code 38194)

### ISSUE THREAD

**Issue by cglacet** [amadeus4dev/developer-guides#6]
When calling **https://test.api.amadeus.com/v1/reference-data/locations/airports** I often end up with the following error.

```json 
{
  "errors": [
    {
      "code": "38194",
      "title": "Too many requests",
      "detail": "The network rate limit is exceeded, please try again later",
      "status": "429"
    }
  ]
}
```

The occurrence of the error seems a bit random to me. Sometimes it shows, but a few seconds after it will not, then it will happen again. 

My test is the following: I launch 6 asynchronous requests (more or less simultaneously) then I wait for all 6 answers to come back. I've looked a bit at the python SDK and I noticed that you do everything in a synchronous way, is that the reason? Or is it possible to achieve what I'm trying to do? Is there a place where all error code are available to us?

**Comment by anthonyroux** [comment]
With the current API Management Platform configuration you can do parallel calls but with a restriction:

- In test: 10 transactions per second (TPS) but each call has to be separated of 100ms
- In production: 20 TPS with 50ms between each call

We are working to improve it and allows a full parallelization without delay between the requests.

This is not linked to the Python SDK.

**Comment by cglacet** [comment]
Ok thanks for the quick update on this. I'll simply add a 100ms sleep between each call to make sure I don't break the limit.

**Comment by anthonyroux** [comment]
And make sure you don't send more than 10 requests per second in test.
If you still have the problem, it might be related to the network latency and you might need to add a bit more than 100ms.

**Comment by anthonyroux** [comment]
I close the issue, let us know if you have any problem

**Comment by cglacet** [comment]
Any update on this issue? Did you implemented a waiting queue or allowed more requests to go through in parallel?

**Comment by donBigote** [comment]
I was having this same issue, and I found your solution here. I used setTimeout in my javascript to created a 100ms window between the requests. I am still getting the "The network rate limit is exceeded, please try again later" 429 error.

I am not sure if I am using setTimeout incorrectly and wanted to share a gist of it to see if you all had any opinion on the matter. 
https://gist.github.com/donBigote/270c9f09653975e6daebefc579a85490

**Comment by anthonyroux** [comment]
Hi, we have updated the rate limits on our APIs. You can find the details [here](https://developers.amadeus.com/self-service/apis-docs/guides/rate-limits-745).

TL;DR:
Still limited in the **test environment** with 10 TPS and a maximum of 1 request every 100ms.
For **production**: 40 TPS (for most of the APIs) without constraints.

**Comment by markduncanawp** [comment]
Hi, we are also experiencing this issue when using /v1/shopping/availability/flight-availabilities in the production environment.  Would this api have the request every 50ms restriction?  If not could you advise other reasons for this error as we are under 40 transactions per second.  However if the 50ms restriction exists then could you advise an approach/technique for developers who work in multi-threaded environments.  Any help is greatly appreciated.
